### ISSUE TITLE
Create posix wrapper for condition variable

### ISSUE THREAD

**Issue by elfenpiff** [eclipse-iceoryx/iceoryx#1037]
## Brief feature description

A posix wrapper for condition variable which allows a safe and easy usage.
* The mutex should be included
* The underlying variable should be part of the condition variable (for the actual condition)
* The predicate should be defined in the constructor
* It should be shared memory compatible

## Detailed information

Such an implementation would allow a safe usage of condition variables throughout the code and could be a replacement of the semaphore based shared memory condition variable implementation.

**Comment by elBoberido** [comment]
In general a good idea but I'm not quite sure the semaphore based shared memory condition variable could be replaced with a mutex based one. AFAIR it was supposed to not have a mutex but a semaphore and a reduced functionality to prevent that crashed apps can leave a locked mutex behing. I think @budrus can give more insight into this.

**Comment by budrus** [comment]
@elfenpiff I do not get this, sorry. We have this condition variable light which has some restrictions compared to a normal condition variable but has the benefit that we do not need a inter-process mutex. Getting rid of these mutexes is always the goal to avoid deadlocks. Here you are discuss the introduction of another mutex? 

cc: @elBoberido

**Comment by elfenpiff** [comment]
@budrus Yes I would like to introduce this for intra process usage. For instance this wrapper would enable us to build a trigger queue and similar constructs with only two lines - and the posix timer demonstrated very well that the bare metal `std::condition_variable` can be used extremely easy wrongly. Furthermore we could never use `std::condition_variable` in a final implementation since it throws exceptions.

But it would not qualify as a replacement for our inter process condition variable and this is also not the intention here.

**Comment by elBoberido** [comment]
> But it would not qualify as a replacement for our inter process condition variable and this is also not the intention here.

Well, the description tells a different story ;)

**Comment by elfenpiff** [comment]
@elBoberido upsi ... the issue was written by another personality of mine ...

**Comment by elBoberido** [comment]
@elfenpiff but wouldn't the proposed `condition_variable` require a predicate and wouldn't this `condition_variable` either always block or never block when the predicate is never changed?
